pkg/apiserver/apiserver.go

```diff
@@ -277,9 +274,13 @@ type StripVersionNegotiatedSerializer struct {
 	runtime.NegotiatedSerializer
 }
 
-func (n StripVersionNegotiatedSerializer) EncoderForVersion(serializer runtime.Serializer, gv unversioned.GroupVersion) runtime.Encoder {
-	encoder := n.NegotiatedSerializer.EncoderForVersion(serializer, gv)
-	return stripVersionEncoder{encoder, serializer}
+func (n StripVersionNegotiatedSerializer) EncoderForVersion(encoder runtime.Encoder, gv unversioned.GroupVersion) runtime.Encoder {
+	serializer, ok := encoder.(runtime.Serializer)
```

Comment:
The overall sig changed, this is a hack because there isn't enough local
info.

On Mon, Apr 25, 2016 at 11:15 AM, David Eads <notifications@github.com>
wrote:

> In pkg/apiserver/apiserver.go
> <https://github.com/kubernetes/kubernetes/pull/24710#discussion_r60930806>
> :
>
> > @@ -277,9 +274,13 @@ type StripVersionNegotiatedSerializer struct {
> >  	runtime.NegotiatedSerializer
> >  }
> >
> > -func (n StripVersionNegotiatedSerializer) EncoderForVersion(serializer runtime.Serializer, gv unversioned.GroupVersion) runtime.Encoder {
> > -	encoder := n.NegotiatedSerializer.EncoderForVersion(serializer, gv)
> > -	return stripVersionEncoder{encoder, serializer}
> > +func (n StripVersionNegotiatedSerializer) EncoderForVersion(encoder runtime.Encoder, gv unversioned.GroupVersion) runtime.Encoder {
> > +	serializer, ok := encoder.(runtime.Serializer)
>
> if you want a serializer, why did you change the sig?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24710/files/e4a1f3c4b34ba5ef0c8f08009e202d0b048498ff#r60930806>
>